api/domain/eventdata/cluster.proto

```diff
@@ -17,44 +17,47 @@ syntax = "proto3";
 // This file follows google's gRPC naming conventions:
 // https://cloud.google.com/apis/design/naming_convention
 
+import "validate/validate.proto";
+
 option go_package = "github.com/finleap-connect/monoskope/pkg/api/domain/eventdata";
 
 package eventdata;
 
 message ClusterCreated {
   // Display name of the cluster
-  string name = 1;
+  string name = 1 [(validate.rules).string.max_bytes = 150];
   // Unique name of the cluster, to be utilized for generating unique labels
   // and symbols, e.g. with metrics.
-  string label = 2;
+  string label = 2 [(validate.rules).string = {pattern: "^[a-zA-Z_]+$", max_bytes: 60}];
   // Address of the KubeApiServer
-  string api_server_address = 3;
+  string api_server_address = 3 [(validate.rules).string.hostname = true];
   // CA certificate of the K8s cluster
   bytes ca_certificate_bundle = 4;
 }
 
 message ClusterCreatedV2 {
   // Unique name of the cluster, to be utilized for generating unique labels
   // and symbols, e.g. with metrics.
-  string name = 1;
+  string name = 1 [(validate.rules).string = {pattern: "^[a-zA-Z_]+$", max_bytes: 60}];
   // Display name of the cluster
-  string display_name = 2;
+  string display_name = 2 [(validate.rules).string.max_bytes = 150];
   // Address of the KubeApiServer
-  string api_server_address = 3;
+  string api_server_address = 3 [(validate.rules).string.hostname = true];
   // CA certificate of the K8s cluster
   bytes ca_certificate_bundle = 4;
 }
 
 message ClusterUpdated {
   // Display name of the cluster
-  string display_name = 1;
+  string display_name = 1 [(validate.rules).string.max_bytes = 150];
   // Address of the KubeApiServer
-  string api_server_address = 2;
+  string api_server_address = 2 [(validate.rules).string.hostname = true];
   // CA certificate of the K8s cluster
   bytes ca_certificate_bundle = 3;
 }
 
 message ClusterBootstrapTokenCreated {
   // JsonWebToken for cluster bootstrap
+  // TODO: dose it make sense to check here it will fail anyway if the token is not valid.
```

Comment:
I would only validate stuff which is sent to the server at the moment. Everything coming from the server is expected to be correct.